src/Features/Core/Portable/Formatting/FormattingDiagnosticAnalyzer.cs

```diff
@@ -35,11 +35,6 @@ private void AnalyzeSyntaxTree(SyntaxTreeAnalysisContext context)
             var cancellationToken = context.CancellationToken;
 
             var options = context.Options.GetDocumentOptionSetAsync(tree, cancellationToken).GetAwaiter().GetResult();
```

Comment:
We are not moving this analyzer to use `GetOption` API?

Reply:
This analyzer has a hard requirement on `Workspace`, but I will update it to use `GetAnalyzerOptionSet` which is a step in the right direction.